server/lib/statsd-collector.js

```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var logger = require('mozlog')('server.statsd');
+var StatsD = require('node-statsd');
+var uaParser = require('ua-parser');
+
+var STATSD_PREFIX = 'fxa.content.';
+
+function StatsDCollector() {
+  var config = require('./configuration');
+  var statsdConfig = config.get('statsd');
+
+  this.host = statsdConfig.host;
+  this.port = statsdConfig.port;
+  this.sampleRate = statsdConfig.sample_rate;
+  this.connected = false;
+  this.client = null;
+}
+
+StatsDCollector.prototype = {
+  /**
+   * Initializes a StatsD socket client
+   */
+  init: function () {
+    var client = this.client = new StatsD(this.host, this.port);
+
+    if (client.socket) {
+      this.connected = true;
+      client.socket.on('error', function (error) {
+        logger.error('Error in stats socket: ', error);
+      });
+    } else {
+      logger.error('StatsD failed to connect to ' + this.host + ':' + this.port);
+      this.connected = false;
+    }
+
+    if (! this.connected) {
+      logger.error('StatsD not connected.');
+    }
+  },
+  /**
+   * Send a formatted metrics object to StatsD
+   *
+   * @param {Object} body
+   */
+  send: function (body) {
```

Comment:
The StdErrorCollector defines a [write](https://github.com/mozilla/fxa-content-server/blob/3fcd448614adb8643e07aae68969b2e61e1ef9b3/server/lib/metrics-collector-stderr.js#L126) function, for consistency let's use the same name.